registry/storage/vacuum.go

```diff
@@ -47,6 +53,99 @@ func (v Vacuum) RemoveBlob(dgst string) error {
 	return nil
 }
 
+// UnlinkLayer removes an entry from repository layers. withDirectory
+// allows to remove <hex-digest> directory containing the link as well.
+func (v Vacuum) UnlinkLayer(repoName string, dgst string, withDirectory bool) error {
```

Comment:
@miminar I am not sure that parent directories can be safely removed for all drivers at request time. It might be okay to allow configuring this for cases when the caller knows that new repositories will not be written during the removal. This could be a field on the `linkedBlobStatter` that could be exposed by an option.